db-functions

```diff
@@ -377,6 +377,31 @@ check_pkgrepos() {
 	return 0
 }
 
+check_stagingrepos() {
+	local pkgfile=${1}
+	local pkgrepo=${2}
+	local pkgname=$(getpkgname "${pkgfile}")
+	local pkgarch=$(getpkgarch "${pkgfile}")
+	local candidate candidates=()
+
+	if in_array "${pkgrepo}" "${STABLE_REPOS[@]}"; then
+		candidates+=($(find_repo_for_package "${pkgname}" "${pkgarch}" "${TESTING_REPOS[@]}"))
+	fi
+	if in_array "${pkgrepo}" "${STABLE_REPOS[@]}" "${TESTING_REPOS[@]}"; then
+		candidates+=($(find_repo_for_package "${pkgname}" "${pkgarch}" "${STAGING_REPOS[@]}"))
+	fi
+	(( ${#candidates[@]} == 0 )) && return 0
+
+	printf '%s\n' "${candidates[@]%-*}"
+	for candidate in "${candidates[@]}"; do
+		for candidate in "${STAGING}/${candidate%-*}"/*${PKGEXTS}; do
```

Comment:
Maybe name the variable in this loop `candidate_pkg` so that we don't reuse the name? I don't think that it matters now, but it might in the future if we ever add code after this loop.